src/components/QrScan/QrScan.js

```diff
@@ -65,18 +73,78 @@ const QrScan = () => {
           setQrCodes(tempScannedCodes);
           history.push('/display-results');
         }
-
         setScannedCodes(tempScannedCodes);
       } else {
         setQrCodes([data]);
         history.push('/display-results');
       }
     }
-  }
+  };
 
   const handleError = () => {
     history.push('/display-results');
```

Comment:
There is a similar warning thrown for `handleError` as the one described in the comment for `handleScan`. 

See https://typeofnan.dev/fix-function-makes-the-dependencies-of-useEffect-hook-change-on-every-render-warning-in-react/.

Instead of moving the function, `handleError` should be wrapped inside a useCallback hook, since it is used in multiple scopes.

Doing something like this should eliminate the warning:
```js
  const handleError = useCallback(() => {
    history.push('/display-results');
  }, [history]);
```